Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -342,10 +340,14 @@ protected override void Dispose(bool disposing)
                 // Prevent an exception from being thrown when disposing of this object
                 logger.Error("Unable to close connection", ex);
             }
-            
-            disposed = true;
 
-            base.Dispose(disposing);
+            // only dispose the connection when it's not pooled

```

Comment:
what will happen when pool wanted to destroy the connection?

Reply:
Good point. Will add fix for such case. Connection pool should be able to unset the pooled flag when removing the connection out of the queue and destroy it.

Reply:
Actually that won't cause issue as the connection status will be closed so Dispose() won't add it to pool again so it will be destroyed.

Reply:
But add reset of the pooled flag anyway to make the status more clear. Also make the change to always call Dispose() do destroy the connection.